pkg/pod/status.go

```diff
@@ -140,7 +147,30 @@ func MakeTaskRunStatus(logger *zap.SugaredLogger, tr v1beta1.TaskRun, pod *corev
 	return *trs, merr.ErrorOrNil()
 }
 
-func setTaskRunStatusBasedOnStepStatus(logger *zap.SugaredLogger, stepStatuses []corev1.ContainerStatus, tr *v1beta1.TaskRun) *multierror.Error {
+func setTaskRunStatusBasedOnSpireVerification(ctx context.Context, logger *zap.SugaredLogger, tr *v1beta1.TaskRun, trs *v1beta1.TaskRunStatus,
+	filteredResults []v1beta1.PipelineResourceResult, spireAPI spire.ControllerAPIClient) {
+
+	if tr.IsSuccessful() && spireAPI != nil &&
+		((tr.Status.TaskSpec != nil && len(tr.Status.TaskSpec.Results) >= 1) || len(filteredResults) >= 1) {
+		logger.Info("validating signed results with spire: ", trs.TaskRunResults)
+		if err := spireAPI.VerifyTaskRunResults(ctx, filteredResults, tr); err != nil {
+			logger.Errorf("failed to verify signed results with spire: %w", err)
+			markStatusSignedResultsFailure(trs, err.Error())
+		} else {
+			logger.Info("successfully validated signed results with spire")
+			markStatusSignedResultsVerified(trs)
+		}
+	}
+
+	// If no results and no results requested, set verified unless results were specified as part of task spec
+	if len(filteredResults) == 0 && (tr.Status.TaskSpec == nil || len(tr.Status.TaskSpec.Results) == 0) {
```

Comment:
Hmm, this feels a bit dangerous, would it be safer to return false here if there were no results to verify?

Reply:
(This is assuming the phase 2 PR which assumes integrity of the Status.TaskSpec.

Reply:
Agree with @lumjjb. As Phase 2 of TEP 0089 would see removal of results by a malicious actor, that would fail the overall TaskRun Status verification. If results are expected but removed, we do have another check for the `RESULT_MANIFEST` that will check that the results requested are returned. If that does not match, it will fail the verification.